erpnext/compliance/doctype/compliance_info/compliance_info.py

```diff
@@ -58,16 +59,34 @@ def get_existing_licensees(license_number, party_type):
 
 
 def validate_license_expiry(doc):
-	if doc.doctype in ("Sales Order", "Sales Invoice", "Delivery Note"):
-		validate_entity_license("Customer", doc.customer)
-	elif doc.doctype in ("Supplier Quotation", "Purchase Order", "Purchase Invoice", "Purchase Receipt"):
-		validate_entity_license("Supplier", doc.supplier)
-	elif doc.doctype == "Quotation" and doc.quotation_to == "Customer":
-		validate_entity_license("Customer", doc.party_name)
+	validate_docinfo = validate_doc_license(doc)
+	if validate_docinfo:
+		if doc.doctype in ("Sales Order", "Sales Invoice", "Delivery Note"):
+			validate_entity_license(doc, "Customer", doc.customer)
+		elif doc.doctype in ("Supplier Quotation", "Purchase Order", "Purchase Invoice", "Purchase Receipt"):
+			validate_entity_license(doc, "Supplier", doc.supplier)
+		elif doc.doctype == "Quotation" and doc.quotation_to == "Customer":
+			validate_entity_license(doc, "Customer", doc.party_name)
 
+@frappe.whitelist()
+def validate_doc_license(doc):
```

Comment:
This is not validating any license in the document if called from the client-side...